_pytest/capture.py

```diff
@@ -252,6 +252,8 @@ def writelines(self, linelist):
         self.write(data)
 
     def __getattr__(self, name):
+        if name == 'name':
+            return repr(self.buffer)
         return getattr(object.__getattribute__(self, "buffer"), name)
```

Comment:
but then buffer is directly usable to begin with, from my pov that call can very likely be shortened to `self.buffer`